judge/models/problem.py

```diff
@@ -407,6 +407,25 @@ def save(self, *args, **kwargs):
 
     save.alters_data = True
 
+    def can_vote(self, user):
+        if not user.is_authenticated:  # reject anons
+            return False
+
+        # if user is unlisted or is banned from submitting to the problem, then they cannot vote
+        if user.profile.is_unlisted or self.banned_users.filter(pk=user.pk).exists():
+            return False
+
+        if user.profile.is_banned_from_voting_problem_points:
+            return False  # site wide voting ban
+
+        if user.profile.current_contest:
+            return False  # if the user is in contest, nothing should be shown
+
+        # the points of all ac submissions in decreasing order
+        ac_sub_points = list(self.submission_set.filter(user=user.profile, result='AC').order_by('-points').values_list('points', flat=True))
```

Comment:
Please use the approach used [here](https://github.com/DMOJ/online-judge/blob/master/judge/utils/problems.py#L37) and just check that `.exists()` is true.

Reply:
I think a better would be to use `.aggregate` to query to the max val,
perhaps it would look something like
```python
best_ac = self.submission_set.filter(user=user.profile, result='AC').aggregate(Max('points'))
return best_ac and best_ac['points'] > self.points - 0.1
```